src/test/services/peers/peer.spec.js

```diff
@@ -0,0 +1,199 @@
+const chai = require('chai');
+
+const expect = chai.expect;
+
+describe('Factory: $peer', () => {
+  let $peer;
+  let $httpBackend;
+  let peer;
+  let conf;
+  let url;
+  const account = { address: '16313739661670634666L' };
+
+  beforeEach(angular.mock.module('app'));
+
+  beforeEach(inject((_$peer_, _$httpBackend_) => {
+    $peer = _$peer_;
+    $httpBackend = _$httpBackend_;
+    conf = {
+      host: 'node07.lisk.io',
+      port: 8000,
+      ssl: true,
+    };
+    peer = new $peer(conf);
+    url = `https://${conf.host}:${conf.port}`;
+  }));
+
+  afterEach(() => {
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  describe('uri', () => {
+    it('contains host:port if port specified', () => {
+      const uri = peer.uri;
+      expect(uri).to.equal(`${peer.host}:${peer.port}`);
+    });
+
+    it('contains host if port not specified', () => {
+      conf.port = undefined;
+      peer = new $peer(conf);
+      const uri = peer.uri;
+      expect(uri).to.equal(`${peer.host}:${peer.port}`);
+    });
+  });
+
+  describe('getPeers()', () => {
+    it('calls peers API and returns promise with peers ', () => {
+      let response;
+      const peers = ['TEST', 'PEER'];
+      const data = {
+        success: true,
+        peers,
+      };
+      $httpBackend.whenGET(`${url}/api/peers?state=2`).respond(200, data);
+      peer.getPeers().then((res) => {
+        response = res;
+      });
+
+      $httpBackend.flush();
+      expect(response).to.deep.equal(peers);
+    });
+  });
+
+  describe('getAccount(address)', () => {
+    it('calls accout API and returns promise with the account', () => {
+      let response;
+      const data = {
+        success: true,
+        account,
+      };
+
+      $httpBackend.whenGET(`${url}/api/accounts?address=${account.address}`).respond(200, data);
+      peer.getAccount(account.address).then((res) => {
+        response = res;
+      });
+
+      $httpBackend.flush();
+      expect(response).to.deep.equal(account);
+    });
+
+    it('calls accout API and id request fails then calls balance API', () => {
```

Comment:
`calls accout` -> `calls account`

`and id request fails` -> `when id request fails`